### ISSUE TITLE
Inverse checker does not consider the meaning

### ISSUE THREAD

**Issue by kachick** [rubocop-hq/rubocop#5085]
Addressing around `Use >= instead of inverting <.`

I can understand the motivation in some objects.

```ruby
!!(1 >= 2) == !(1 < 2) #=> true
```

But is this an intentional behavior?

```ruby
!!(Integer >= Enumerable) == !(Integer < Enumerable) #=> false
```

I think the style restriction will not fit for ruby `Comparable`.

---

## Expected behavior

Does not any alert

## Actual behavior

Alerting `Use >= instead of inverting <.` as below

```
Offenses:

lib/aya/coaching/tmp.rb:1:30: C: Use >= instead of inverting <.
!!(Integer >= Enumerable) == !(Integer < Enumerable) #=> false
                             ^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem

```shell
rubocop example.rb 
```

example.rb 
```ruby
!!(Integer >= Enumerable) == !(Integer < Enumerable) #=> false
```

## RuboCop version

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.4.2 x86_64-darwin16)
```

**Comment by mikegee** [comment]
I suppose the cop could check if both operands are camel-case constants (assuming that means they are classes).

**Comment by kachick** [comment]
Will it work in below case?

```ruby
integer = Integer
enumerable = Enumerable
  
!!(integer >= enumerable) == !(integer < enumerable)
```

And in my understanding, we expect `Comparable` returns `nil` in uncomparable relations. It is not scoped in class/module objects.

**Comment by pocke** [comment]
> I suppose the cop could check if both operands are camel-case constants (assuming that means they are classes).

👍 



> Will it work in below case?


Unfortunately the cop cannot avoid the case currently. It's a limitation of RuboCop.
(note: If we implement tracing local variable, the cop can avoid it. But if the `integer` is a method call, avoiding it is completely impossible.)

---


I think we should change three things.
First, we should change the cop to check camel-case constants ( @mikegee 's suggestion)
By this change, the cop can process correctly almost cases.
Second, we should probably change the offense message. The cop can describe about this problem in offense messages.
Third, we should probably disable the auto-correction.

----


RuboCop is not completely. In other words, RuboCop has false positives. I think if a cop works correctly in almost cases, we should allow false positives. But we should treat false positives carefully. For example, describing false positives in messages and/or disabling auto-correction.

**Comment by kachick** [comment]
Sure. Thanks for the sharing! 🙏

**Comment by rrosenblum** [comment]
Would it be beneficial to move the checks for `<`, `>=`, `>`, and `<=` out of the default set of checked methods? 

Please take into consideration #4289. It is a request to be able to disable the default set of inverse methods. I know it isn't directly related, but it seems like there could be a similar solution with being able to configure the methods that are checked.

> note: If we implement tracing local variable, the cop can avoid it.

I would love to have this functionality. I have mentioned the value that it could provide in the past. I somewhat feel like this is almost a request to build type checking into Ruby (obviously an exaggeration).

**Comment by bbatsov** [comment]
One more cop for us to mark as `unsafe`. //cc @Darhazer
